File: test/benchmark/cases/int_loop/case.java
class Main {
    public static void main(String[] args) {
        var raw = System.getenv("CP_BENCH_INPUT");
        if(raw == null) {
            System.exit(1);
        }

        var limit = Long.parseLong(raw);
        var index = 0L;
        var total = 7L;

        while(index < limit) {
            total = (total + ((index * 31 + 17) % 1009)) % 1000003;
            index += 1;
        }

        System.exit((int)(total % 251));
    }
}

File: test/benchmark/cases/int_loop/case.cpp
#include <cstdint>
#include <cstdlib>

auto main() -> int
{
    auto const* raw = std::getenv("CP_BENCH_INPUT");
    if(raw == nullptr) {
        return 1;
    }

    auto const limit = std::atoi(raw);
    auto index = std::int64_t{ 0 };
    auto total = std::int64_t{ 7 };

    while(index < static_cast<std::int64_t>(limit)) {
        total = (total + ((index * 31 + 17) % 1009)) % 1000003;
        index += 1;
    }

    return static_cast<int>(total % 251);
}
